pkg/ddc/jindofsx/ufs_internal.go

```diff
@@ -86,14 +86,8 @@ func (e *JindoFSxEngine) mountUFS() (err error) {
 			mount.MountPoint = "local://" + ufsVolumesPath
 		}
 		if !mounted {
-			if mount.Path != "" {
-				err = fileUitls.Mount(mount.Path, mount.MountPoint)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			err = fileUitls.Mount(mount.Name, mount.MountPoint)
+			mountPathInJindo := utils.UFSPathBuilder{}.GenUFSPathInUnifiedNamespace(mount)
+			err = fileUitls.Mount(mountPathInJindo, mount.MountPoint)
 			if err != nil {
 				return err
 			}
```

Comment:
A corner case of backward incompatibility is not handled here. Previous implementation allows users to define `mount.Path` without a prefix "/", and `JindoEngine` passes `mount.Path` to `JindoFileUtils.Mount()`. Inside `JindoFileUtils.Mount()`, the function automatically add a prefix "/" to `mount.Path` and try to mount on it.

That is to say, if a user used to set `Dataset.spec.mounts[*].path` to a string without a prefix "/", he or she may see different behavior  before and after this PR is merged. For example:
```
apiVersion: data.fluid.io/v1alpha1
kind: Dataset
metadata:
  name: demo2
spec:
  placement: Shared
  accessModes:
  - ReadOnlyMany
  mounts:
      mountPoint: <mountPoint>
      name: demo2 # After this pr is merged, mountPoint is mounted to /demo2
      path: mybucket # Before this pr is merged, mountPoint is mounted to /mybucket
```